### ISSUE TITLE
Sending an invalid stream keys payload to the admin API causes issues with the admin

### ISSUE THREAD

**Issue by breedx2** [owncast/owncast#3082]
### Share your bug report, feature request, or comment.

Firstly, thanks for a killer piece of software. This helped us build friendships and survive during the pandemic. :pray: 

This started with `0.1.0` and took a bit of troubleshooting to narrow down what combination of things was causing this. It looks like `POST`ing a new key to `/api/admin/config/streamkeys`  breaks both streaming and the admin UI ( Configuration -> Server Setup -> Stream Keys).

Steps to reproduce:

1. `docker run -it --rm -v /tmp/owncast-data:/app/data -p 8089:8080 gabekangas/owncast:0.1.0`
2. `curl -i -X POST -H 'Content-type: application/json' --data '{"id":"zz","comment":"ouch"}' http://admin:abc123@<host>:8080/api/admin/config/streamkeys` - note this returns a 200, which also indicates that the complexity requirements are also being bypassed, but let's do that in a different issue...
3. Navigate to `http://admin:abc123@<host>/admin/config/server/` and then click the "Stream Keys" sub tab. The UI yields an error: `Application error: a client-side exception has occurred (see the browser console for more information).`. I have also confirmed that this also breaks new streaming inputs, even to the default stream key.

I've also tried this with more complex/secure stream keys but it yields the same behavior.

It's not super relevant to the issue, but I thought I'd mention that I've been using ansible to stand up and configure owncast, and some of that is leveraging the REST apis, which is how I found this. Thanks again!

**Comment by gabek** [comment]
The stream keys API should be a `value` property with an array of keys and comments. For example:

```json
{"value":[{"key":"abc123","comment":"Default stream key"},{"key":"5M^cVUeC&n$Gn7rq*Er9@4dQG2zwfK","comment":"My new key"},{"key":"8ndq3ZsolC^3$dm5FtKogPuCjGZ4&A","comment":"My new key"}]}
```

It looks like what you're trying to send is incorrect. However, the problem is it doesn't catch your incorrect payload.

**Comment by breedx2** [comment]
Hey @gabek thanks for responding. In REST I'd normally expect to POST a single entity into a collection CRUD style, but if I was replacing the entire collection I'd expect a PUT, and even more confusing are the PATCH use cases....heh. Whatever.

All that aside, I'm now sure you're right about my request not being correct now that owncast supports multiple streamkeys. Looking at the API docs, though:

![image](https://github.com/owncast/owncast/assets/1888255/80d841ad-8ba5-409b-918b-b4eb3913aaf3)

I'm happy to change my POST to send an array, but is the json key actually `key` as you put above, or is it `id` like the docs have? If it's `key`, should I file a separate issue on the docs?

Thanks again!

**Comment by gabek** [comment]
I admit the API documentation is bad, really bad. It's hand written by me, and when something changes or I change my mind about something I have to go and remember to manually change it. I think at this point I should remove all internal APIs from the documentation to solve problems like this. APIs, like this stream keys one, wasn't built to be used by external tooling. External APIs, of course, deserve extra attention and documentation, but the internal ones create so much manual work for me to keep them up to date.

In short, the documentation is wrong, what is being sent by the admin is correct.

**Comment by breedx2** [comment]
> I admit the API documentation is bad, really bad. It's hand written by me, and when something changes or I change my mind about something I have to go and remember to manually change it. I think at this point I should remove all internal APIs from the documentation to solve problems like this. APIs, like this stream keys one, wasn't built to be used by external tooling. External APIs, of course, deserve extra attention and documentation, but the internal ones create so much manual work for me to keep them up to date.
> 
> In short, the documentation is wrong, what is being sent by the admin is correct.

Thanks, I can appreciate the work required to keep both apis and their docs up to date. In your defense, the docs look so good and are so comprehensive that I assumed they were autogenerated from source (honestly, hadn't looked).

Anyway, I can respect not wanting to publish docs on "internal" APIs, but having a way to automate changing the default stream key is invaluable. Especially on new installs. I would make a case for this being first-class functionality for those of us who do recurring short-lived installs who don't want the default stream key exposed for long. Happy to continue in the other issue if this is getting too long/offtopic.

**Comment by gabek** [comment]
I'm not sure if it's helpful in this case, but there is a `--streamkey` command line flag that will set a single stream key, and it will use that instead of any others as long as that's provided on the command line.

```
  -streamkey string
    	Set a temporary stream key for this session
```

**Comment by breedx2** [comment]
:sunglasses: That's helpful. :pray:  In my case it's a trade-off between locally templating a startup script or leveraging ansible to REST up the streamkeys. Pros and cons for both options, as always. I only need one stream key, so the commandline arg is a very reasonable option. The existing use of ansible to REST API all the other server configs kinda suggests I should just fix my REST payload and move on....at least until it changes again. :wink:

**Comment by gabek** [comment]
> In your defense, the docs look so good and are so comprehensive that I assumed they were autogenerated from source (honestly, hadn't looked).

I've tried to make that happen, but the tooling just doesn't exist for OpenAPI 3. Lots of tooling for OpenAPI 2/Swagger, however. The other option is to rebuild the routing layer with a framework such as Gin that has that feature built-in, but I don't want to rewrite a major layer with a bulky framework I don't need just for documentation. So I've just written it manually. When I remember.
